src/ui/public/agg_types/metrics/top.js

```diff
@@ -0,0 +1,59 @@
+import { noop, get } from 'lodash';
+import MetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import topEditor from 'ui/agg_types/controls/top.html';
+
+export default function AggTypeMetricTopProvider(Private) {
+  const MetricAggType = Private(MetricAggTypeProvider);
+
+  return new MetricAggType({
+    name: 'top_hits',
+    title: 'Top',
+    makeLabel: function (aggConfig) {
+      const prefix = aggConfig.params.sortOrder.val === 'desc' ? 'Latest' : 'Earliest';
+      return `${prefix} ${aggConfig.params.field.displayName}`;
+    },
+    params: [
+      {
+        name: 'field',
+        write(agg, output) {
+          output.params = {
+            size: 1,
+            _source: agg.params.field.name
```

Comment:
You probably want to set [docvalue_fields](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-request-docvalue-fields.html) instead of `_source`. It would be more consistent with the way the other aggregations work and also possibly more performant. Down the road, once we know [which fields are aggregatable/searchable](https://github.com/elastic/elasticsearch/pull/17980) we could enhance this `top` metric to use `_source` vs `docvalue_fields` depending on what kind of field is selected.